### PR TITLE
Add Ukrainian translation v1.5.2

### PR DESCRIPTION
**Pull request by drtwisted**
**NOTE:** there's an issue with `Glasstown_NBP.ttf` font. Lower case latter 'є' is mirrored and looks like `э`. You shouldn't confuse it with actual `э` which is something that exists.

Below is the actual code of the symbol. And the way it should look.
image

I can provide the corrected font.

### CODE REVIEW

File: ukrainian.ini [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+Exit="Вийти з гри"
+Tip1="| - Вибрати"
+Tip2="| - Назад"
+GameSlot_NewGame="Нова гра"
+GameSlot_Easy="Легко"
+GameSlot_Normal="Нормально"
+GameSlot_Hard="Складно"
+GameSlot_Continue="Продовжити гру"
+GameSlot_Copy="Скопіювати дані"
+GameSlot_Erase="Вилучити дані"
```
Comment (kirillstarkov) [inline ukrainian.ini:23]: Видалити дані

File: ukrainian.ini [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+GameSlot_EraseConfirm="Вилучити дані гри?"
```
Comment (kirillstarkov) [inline ukrainian.ini:24]: Видалити дані гри

File: ukrainian.ini [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+[GlobalOptions]
+Yes="Так"
+No="Ні"
+Enabled="Увімкнено"
+Disabled="Вимкнено"
+Exit="Гаразд"
+Exit_Tip="Зберегти поточні налаштування та вийти"
+ExitMain_Tip="Зберегти та перейти до головних налаштувань"
+Defaults="Типово"
+Defaults_Tip="Повернути до типових налаштувань"
```
Comment (kirillstarkov) [inline ukrainian.ini:41]: Повернути до стандартних налаштувань

File: ukrainian.ini [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+
+[OptionsMain]
+Title="Налаштування"
+Display="Налаштування екрана"
+Display_Tip="Змінити налаштування екрана"
+Sound="Налаштування звуку"
+Sound_Tip="Регулювання гучності звуків та музики"
+Control="Налаштування керування"
+Control_Tip="Налаштування способів керування, клавіатури та ґеймпада"
```
Comment (kirillstarkov) [inline ukrainian.ini:50]: геймпада

File: ukrainian.ini [hunk 5]
```diff
@@ -0,0 +1,666 @@
...
+Resume_Tip="Продовжити гру"
+Restart="Почати з останнього збереження"
+Restart_Tip="Полишити поточну гру та завантажити останнє збереження"
+Options="Налаштування"
+Options_Tip="Змінити налаштування гри"
+Quit="Вийти з гри"
+Quit_Tip="Полишити поточну гру та вийти до головного меню"
+
+[QuitMenu]
+Exit="Вихід — ви певні?"
```
Comment (kirillstarkov) [inline ukrainian.ini:210]: Вихід — ви впевнені?

File: ukrainian.ini (hunk 1 above)
Comment (drtwisted) [reply ukrainian.ini:23]: Це синонімічні конструкції, не бачу сенсу в заміні.

File: ukrainian.ini (hunk 2 above)
Comment (drtwisted) [reply ukrainian.ini:24]: Це синонімічні конструкції, не бачу сенсу в заміні.

File: ukrainian.ini (hunk 3 above)
Comment (drtwisted) [reply ukrainian.ini:41]: Це синонімічні конструкції, не бачу сенсу в заміні.

File: ukrainian.ini (hunk 4 above)
Comment (drtwisted) [reply ukrainian.ini:50]: Це синонімічні конструкції, не бачу сенсу в заміні.

File: ukrainian.ini (hunk 5 above)
Comment (drtwisted) [reply ukrainian.ini:210]: Це синонімічні конструкції, не бачу сенсу в заміні.

**Comment by Miepee** [comment]
Resolved conversations, trusting drtwisted here more, and from my limited knowledge seem to agree with him here.

File: ukrainian.ini [hunk 6]
```diff
@@ -0,0 +1,666 @@
...
+SensitivityMode_Tip = "Зменшує інтенсивність деяких ефектів освітлення"
+Widescreen = "Широкий екран (ШЕ)"
+Widescreen_Tip = "Перемикає ШЕ. Для застосування у грі, потрібно перейти між кімнатами"
+
+[ExtrasMenu]
+Extras = "Додаткове"
+Extras_Tip = "Додатковий вміст"
+Title_Extras = "Додаткове"
+Controller = "Підказки контролера"
+Controller_Tip = "Задає тип контролера, кнопки якого буде показано в підказках на екрані"
```
Comment (Miepee) [inline ukrainian.ini:585]: This clips in 4:3
grafik

File: ukrainian.ini [hunk 7]
```diff
@@ -0,0 +1,666 @@
...
+AimingStyle = "Спосіб прицілювання"
+AimingStyle_Tip = "Перемикає прицілювання на стиль «Zero Mission» або «Super Metroid»"
+AimingStyle_1Button = "1 Кнопка"
+AimingStyle_1Button_Tip = "Натисніть [Вгору] або [Вниз] під час прицілювання для вибору напрямку"
+AimingStyle_2Buttons = "2 Кнопки"
+AimingStyle_2Buttons_Tip = "Прицілювання по діагоналі за допомогою 2-х окремих кнопок"
+MissileStyle = "Спосіб вибору ракет"
+MissileStyle_Tip = "Перемикає спосіб вибору зброї між «Zero Mission» та «Super Metroid»"
+MissileStyle_Toggle = "Перемикання"
+MissileStyle_Toggle_Tip = "Ракети активовано за замовчуванням. Для деактивації треба натиснути знову"
```
Comment (Miepee) [inline ukrainian.ini:104]: This clips in 4:3
grafik

File: ukrainian.ini [hunk 8]
```diff
@@ -0,0 +1,666 @@
...
+TrooperLogs_Text7 = "Рядовий 1-го класу М. Гадсон# #Ми отримали сигнал лиха перш ніж той жукоподібний метроїд підсмажив комунікаційну антену, але проблема у капітані. Він хоче пошукати підземну команду. Нам там щойно надерли сраку, а він хоче туди повернутися? Я маю кращу ідею, як щодо «накивати п’ятами від них та від цієї планети»? Мені біля сраки, якщо мене називатимуть боягузом чи зачинять на нарах із піратами в Штабі, я просто хочу бути якнайдалі від цієї кам’яної смертельної пастки."
+TrooperLogs_Text8 = "Фахівець Ф. Мак# #Табору немає. Блейна теж. Ми бігли до ліфта, коли медузоподібний метроїд вхопив його. Довбень увіпхнув руйнівну бомбу в його рота; наступної миті мене збило з ніг, ліфт укритий друзками каміння, а метроїд невимушено летить назад. Трясця, Блейне, наші технології не працюють так, як технології Аран!#Але я не спускав з нього очей. Вистежив його доколи він влігся та змінив колір. ##Тварюко, час відплати…"
+
+[DPadMenu]
+DPadRebind = "Перепризначити напрямки"
+DPadRebind_Tip = "Дозволяє перепризначення напрямки рухів на окремі кнопки"
```
Comment (Miepee) [inline ukrainian.ini:652]: These clip on 4:3
grafik

File: ukrainian.ini [hunk 9]
```diff
@@ -0,0 +1,666 @@
...
+GBred = "Віртуально-троїдний"
+GBcustom = "Користувацький"
+IGT = "Показувати ВІТ"
+IGT_Tip = "Показувати ігровий таймер на користувацькому інтерфейсі"
+
+[GameplayMenu]
+Gameplay = "Ігролад"
+Title_Gameplay = "Налаштування ігроладу"
+CustomSeed = "Ключ"
+CustomSeed_Tip = "Визначає розташування речей у «Випадкова гра+»"
```
Comment (Miepee) [inline ukrainian.ini:620]: This clips on 4:3
grafik

File: ukrainian.ini [hunk 10]
```diff
@@ -0,0 +1,666 @@
...
+Ex = "Below are all the fields that were added to the language file in the updates."
+
+[Title-Additions]
+GameSlot_NewGame_NormalGame = "Звичайна гра"
+GameSlot_NewGame_NewGamePlus = "Нова гра+"
+GameSlot_NewGame_RandomGamePlus = "Випадкова гра+"
+GameSlot_NewGame_ClassicMode = "Класичний режим"
+GameSlot_NewGame_PowerupsOnly = "Лише підсилення"
+GameSlot_NewGame_StartingPaths = "Початкові навички"
+GameSlot_NewGame_SplitRandom = "Випадкове розташування підсилень"
```
Comment (Miepee) [inline ukrainian.ini:459]: This is a tad long. Any way of shortening it?
grafik

File: ukrainian.ini [hunk 11]
```diff
@@ -0,0 +1,666 @@
...
+MorphBall = "МОРФКУЛЬКА"
+SpiderBall = "ПАВУКУЛЬКА"
+SpringBall = "СТРИБОКУЛЬКА"
+Bombs = "БОМБИ"
+PowerGrip = "СИЛОВИЙ ХВАТ"
+ScrewAttack = "ГВИНТОВА АТАКА"
+Boots = "ЧОБОТИ"
+HiJumpBoots = "ВИСОКОСТРИБНІ ЧОБОТИ"
+SpaceJump = "КОСМІЧНИЙ СТРИБОК"
+SpeedBooster = "ПРИШВИДШЕННЯ"
```
Comment (Miepee) [inline ukrainian.ini:484]: A bunch of these clip.
grafik

File: ukrainian.ini [hunk 12]
```diff
@@ -0,0 +1,666 @@
...
+SelectLog_Tip = " - Вибрати дієпис"
+ExpandLog_Tip = "| - Розгорнути дієпис"
+ScrollText_Tip = " - Гортати текст"
+Back_Tip = "| - Назад"
+
+[Equipment]
+PowerSuit_Tip = "Звичайний броньований костюм Самус."
+VariaSuit_Tip = "Додатково захищає від ворогів та небезпечних оточень."
+GravitySuit_Tip = "Захищає від усіх небезпек оточення та усуває спротив у рідинах."
+ChargeBeam_Tip = "Затисніть |, щоб сконцентрувати енергію променя. Відпустіть, щоб вистрілити."
```
Comment (Miepee) [inline ukrainian.ini:235]: This text clips on 4:3
grafik

File: ukrainian.ini [hunk 13]
```diff
@@ -0,0 +1,666 @@
...
+MissileTank = "Отримано капсулу ракет"
+SuperMissileTank = "Отримано капсулу надракет"
+PowerBombTank = "Отримано капсулу силових бомб"
+ScanningBioform = "Сканування біоформи"
+ScanningMechanism = "Сканування механізму"
+ScanningEnvironment = "Сканування оточення"
+
+[IntroCutscene]
+Scene1 = "Метроїди. #Численні інциденти в минулому#виявили надзвичайно руйнівну силу#цих форм життя, що ширять#страх всіма відомими планетарними системами."
+Scene2 = "З надією встановити мир в галактиці, #Галактична Федерація наказує#винищити весь вид метроїдів.#Мене було обрано з поміж інших для цієї місії."
```
Comment (Miepee) [inline ukrainian.ini:446]: This text doesn't full yfit on 4:3

File: ukrainian.ini [hunk 14]
```diff
@@ -0,0 +1,666 @@
...
+DiagonalAiming_Text = "Утримуйте | , щоб прицілитись по діагоналі"
+MorphBall = "Морфкулька"
+MorphBall_Text = "Тисніть | коли присіли, щоб морфувати в кульку"
+AimingDownwards = "Прицілювання вниз"
+AimingDownwards_Text1 = "Тицьніть | , доки утримуєте | , щоб прицілитись вниз по діагоналі"
+AimingDownwards_Text2 = "Утримуйте | , щоб прицілитись вниз по діагоналі"
+ShootingMissiles = "Стрільба ракетами"
+ShootingMissiles_Text1 = "Натисніть | , щоб вибрати ракети"
+ShootingMissiles_Text2 = "Утримуйте | , щоб увімкнути ракети"
+Autoad = "Autoad"
```
Comment (Miepee) [inline ukrainian.ini:383]: is this intended to not be translated?

**Review by Miepee — COMMENTED** [review]
Found some things.

**Comment by Miepee** [comment]
Authors are too long
grafik

File: ukrainian.ini (hunk 14 above)
Comment (drtwisted) [reply ukrainian.ini:383]: Oh well, this actually is a mystery to me, as for where it is used and what does it mean. Could you please clarify?

File: ukrainian.ini (hunk 11 above)
Comment (drtwisted) [reply ukrainian.ini:484]: I'll be able to fix the costume related ones, but not the Misc & Boots section, it just would not make sense, unfortunately.

File: ukrainian.ini (hunk 10 above)
Comment (drtwisted) [reply ukrainian.ini:459]: Done my best.

File: ukrainian.ini (hunk 11 above)
Comment (Miepee) [reply ukrainian.ini]: Are you not being able to shorten them by using `.` as a shortener?
There has to be a way to shorten those too. The portuguese brazilian translation found themselves in the same issue and then managed to fix those by changing some words and using abbreviations.

File: ukrainian.ini (hunk 14 above)
Comment (Miepee) [reply ukrainian.ini]: Seems unused actually.

File: ukrainian.ini [hunk 15]
```diff
@@ -232,7 +232,7 @@ Back_Tip="| - Назад"
 PowerSuit_Tip="Звичайний броньований костюм Самус."
 VariaSuit_Tip="Додатково захищає від ворогів та небезпечних оточень."
 GravitySuit_Tip="Захищає від усіх небезпек оточення та усуває спротив у рідинах."
-ChargeBeam_Tip="Затисніть |, щоб сконцентрувати енергію променя. Відпустіть, щоб вистрілити."
+ChargeBeam_Tip="Затисніть |, щоб сконцентрувати енергію. Відпустіть, для пострілу."
```
Comment (Miepee) [inline ukrainian.ini:235]: It still clipsgrafik

File: ukrainian.ini [hunk 16]
```diff
@@ -0,0 +1,666 @@
...
+MorphBall="МОРФКУЛЬКА"
+SpiderBall="ПАВУКУЛЬКА"
+SpringBall="СТРИБОКУЛЬКА"
+Bombs="БОМБИ"
+PowerGrip="СИЛОВИЙ ХВАТ"
+ScrewAttack="ГВИНТОВА АТАКА"
+Boots="ЧОБОТИ"
+HiJumpBoots="ВИСОКОСТРИБНІ ЧОБОТИ"
+SpaceJump="КОСМІЧНИЙ СТРИБОК"
+SpeedBooster="ПРИШВИДШЕННЯ"
```
Comment (Miepee) [inline ukrainian.ini:484]: These four sadly still clip (see above).
Maybe get some inspiration from other russian translations in order to shorten these?

File: ukrainian.ini [hunk 17]
```diff
@@ -0,0 +1,666 @@
...
+UpdateLog_Text="Натисніть | , щоб відкрити книгу дієпису"
+UpdateLog_Start="Почати"
+Monsters="Сканування енергетичної схеми"
+Monsters_Text="Метроїдів виявлено у цій місцевості"
+
+[SaveLocation]
+Save00="Місце посадки"
+Save01="Поверхнева печера"
+Save02="Поверхнева печера"
+Save03="Селекційні майданчики Золотого храму"
```
Comment (Miepee) [inline ukrainian.ini:402]: Clips
grafik

File: ukrainian.ini [hunk 18]
```diff
@@ -0,0 +1,666 @@
...
+Save10="Вхід до заводу роботів"
+Save11="Сховище заводу роботів"
+Save12="Видобувний об’єкт"
+Save13="Західна печера"
+Save14="Вхід у вежу"
+Save15="Околиці вежі"
+Save16="Геотермальний завод"
+Save17="Вхід до розподільчого центру"
+Save18="Головний розподільний об’єкт"
+Save19="Сховище розподільчого об’єкта"
```
Comment (Miepee) [inline ukrainian.ini:418]: These clip too

**Review by Miepee — COMMENTED** [review]
Last things I found, will merge when these are fixed.

File: ukrainian.ini (hunk 18 above)
Comment (drtwisted) [reply ukrainian.ini:418]: I've shortened the naming of the difficulty as well, so it should aid other combinations.

**Comment by Miepee** [comment]
Looks good now!

**Pull request merged**